zad2-mdarul/main.py

```
@@ -0,0 +1,174 @@
+# -*- coding: utf-8 -*-
+import os
+import csv
+import argparse
+
+
+class ParserCollisionsExceptions(Exception):
+    """Cant be lower than 0"""
+    pass
+
+
+def parser_decorator(function):
+    def communiacte():
+        print("Parsing arguments...")
+        return function()
+    return communiacte
+
+
+@parser_decorator
+def parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("path", type=str, help="directory with enroll .csv files")
+    parser.add_argument("group", type=str, help='term of the group in format "day week_type time_start - time_end "')
+    parser.add_argument("subject", type=str, help="name of the subject (the same as file name)")
+    parser.add_argument("max_collisions", type=int, help="max number of students with collisions")
+    args = parser.parse_args()
+    path = args.path
+    group = args.group
+    subject = args.subject
+    try:
+        max_collisions = args.max_collisions
+        if max_collisions < 0:
+            raise ParserCollisionsExceptions
+    except ParserCollisionsExceptions:

```

You should either let it be thrown or handle it differently. Now you print messages and then return (possibly invalid) data anyways :D